drivers/soundwire/bus.c

```diff
@@ -1404,7 +1406,7 @@ static int sdw_handle_port_interrupt(struct sdw_slave *slave,
 		count++;
 
 		/* we can get alerts while processing so keep retrying */
-	} while (status != 0 && count < SDW_READ_INTR_CLEAR_RETRY);
+	} while ((status & SDW_DPN_INTERRUPTS) && (count < SDW_READ_INTR_CLEAR_RETRY));
```

Comment:
ditto

Reply:
I don't like having to think about operator precedence between a & b && c < d, and if we use parenthesis on one part of the formula we do in the second as well. no change.